tests/test_uncrustify/test.py

```diff
@@ -176,7 +193,10 @@ def build_from_declaration(self, decl, group):
         else:
             self.test_lang = test_dir
 
-        if parts[0].endswith('!'):
```

Comment:
I'm okay with this. However, this could also be rewritten:
```python
m = re.match(r'(\d+)([~!]*)', parts[0]
num = m.group(1)
is_rerun = ('!' in m.group(2))
is_xfail = ('~' in m.group(2))
```

...which do you like better?